lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -280,6 +280,73 @@ class Class < Namespace
       class Constant < Entry
       end
 
+      class Parameter
+        extend T::Helpers
+        extend T::Sig
+
+        abstract!
+
+        sig { returns(Symbol) }
+        attr_reader :name
+
+        sig { params(name: Symbol).void }
+        def initialize(name:)
+          @name = name
+        end
+      end
+
+      class RequiredParameter < Parameter
+      end
+
+      class Method < Entry
```

Comment:
Should we add a flag for instance vs. singleton method?

Reply:
We've updated to use separate classes to distinguish each.